superset/config.py

```diff
@@ -986,6 +989,9 @@ def CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC(  # pylint: disable=invalid-name
 # See here: https://github.com/dropbox/PyHive/blob/8eb0aeab8ca300f3024655419b93dad926c1a351/pyhive/presto.py#L93  # pylint: disable=line-too-long,useless-suppression
 PRESTO_POLL_INTERVAL = int(timedelta(seconds=1).total_seconds())
 
+# A list of custom Trino authentication that allowed.
+TRINO_EXTRA_AUTHENTICATION: List[str] = []
```

Comment:
To avoid cluttering the config with engine-specific props, could we make this similar to how `TIME_GRAIN_ADDON_EXPRESSIONS` is defined, where we have a `dict` with the engine name as a key. Something like

```python
ALLOWED_EXTRA_AUTHENTICATIONS: Dict[str, List[str]] = {}
```

Then when you implement these, it would be

```python
ALLOWED_EXTRA_AUTHENTICATIONS: Dict[str, List[str[]] = {
    "trino": [
        "<your.module:AuthClass>",
        "<another.extra:auth_method>",
    ],
}
```

Reply:
@villebro changed as your suggestion